imap_processing/swe/l1b/swe_l1b_science.py

```diff
@@ -134,51 +134,108 @@ def convert_counts_to_rate(data: np.ndarray, acq_duration: int) -> npt.NDArray:
+    is in-flight calibration data that covers science data.
 
-    What this function is doing:
+    TODO: decide filename convention given this information. This function
+    is a placeholder for reading in the calibration data until we decide on
+    how to read calibration data through dependencies list.
 
-    | 1. **Reading Calibration Data**: The function first reads a file containing
-    |     calibration data for electron measurements over time. This data helps
-    |     adjust or correct the measurements based on changes in the instrument's
-    |     sensitivity.
+    Returns
+    -------
+    in_flight_cal_df : pandas.DataFrame
+        DataFrame with in-flight calibration data.
+    """
+    # TODO: Read in in-flight calibration file.
 
-    | 2. **Interpolating Calibration Factors**: Imagine you have several points on
-    |     a graph, and you want to estimate values between those points. In our case,
-    |     these points represent calibration measurements taken at different times.
-    |     The function figures out which two calibration points are closest in time
-    |     to the specific measurement time you're interested in.
+    # Define the column headers
+    columns = ["met_time", "cem1", "cem2", "cem3", "cem4", "cem5", "cem6", "cem7"]
 
-    | 3. **Calculating Factors**: Once it finds these two nearby calibration points,
-    |     the function calculates a correction factor by drawing a straight line
-    |     between them (linear interpolation). This factor helps adjust the measurement
-    |     to make it more accurate, considering how the instrument's sensitivity changed
-    |     between those two calibration points.
+    # Create an empty DataFrame with the specified columns
+    empty_df = pd.DataFrame(columns=columns)
+    return empty_df
 
-    | 4. **Returning the Correction Factor**: Finally, the function returns this
-    |     correction factor. You can then use this factor to adjust or calibrate your
-    |     measurements at the specific time you're interested in. This ensures that
-    |     your measurements are as accurate as possible, taking into account the
-    |     instrument's changing sensitivity over time.
+
+def calculate_calibration_factor(
+    acquisition_times: np.ndarray, cal_times: np.ndarray, cal_data: np.ndarray
+) -> npt.NDArray:
+    """
+    Calculate calibration factor using linear interpolation.
+
+    Steps to calculate calibration factor:
+        1. Convert input time to match time format in the calibration data file.
+           Both times should be in S/C MET time.
+        2. Find the nearest in time calibration data point.
+        3. Linear interpolate between those two nearest time and get factor for
+           input time.
 
     Parameters
     ----------
-    time : int
-        Input time.
+    acquisition_times : numpy.ndarray
+        Data points to interpolate. Shape is (24, 30, 7).
```

Comment:
Didn't you drop the 7 from the times variable? Is it also adding a sweep_number to the front too? `(n_sweeps, 24, 30)`
```suggestion
        Data points to interpolate. Shape is (24, 30).
```